restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataValidator.java

```diff
@@ -286,15 +295,38 @@ public void validate(ValidatorContext context)
     {
       super.validate(context);
       DataElement element = context.dataElement();
-      if (_readOnlyRestrictedPredicate.evaluate(element))
+      System.out.println("Validate: " + element);
+      if (_readOnlyRestrictedPredicate.evaluate(element) && !grantArrayDescendantException(element))
```

Comment:
why not 
```
if (!grantArrayDescendantException(element)) {
  // Original logic
}
```

Reply:
Since this method traverses the data tree, it can potentially have performance issues. If we did it your way, it would be invoked for each element in the tree. Since my way is only invoked when the predicate is matched, this means the method will only be invoked for `ReadOnly`/`CreateOnly` elements that are mistakenly included (or those which are included as part of an array).

Reply:
Is this println by design or by choice? Logging maybe better?

Reply:
@karthikrg It was just an accident, I added this during testing to see the the order in which elements in the tree are validated. Given that this is invoked once for each element in a JSON object, I think it would flood the logs.